packages/custom-objects-exporter/src/main.js

```diff
@@ -0,0 +1,117 @@
+/* @flow */
+import { createClient } from '@commercetools/sdk-client'
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import {
+  createAuthMiddlewareForClientCredentialsFlow,
+  createAuthMiddlewareWithExistingToken,
+} from '@commercetools/sdk-middleware-auth'
+import { createUserAgentMiddleware } from '@commercetools/sdk-middleware-user-agent'
+import JSONStream from 'JSONStream'
+import type {
+  ApiConfigOptions,
+  ExporterOptions,
+  LoggerOptions,
+} from '../../../types/customObjects'
+import type { Client, ClientRequest } from '../../../types/sdk'
+import pkg from '../package.json'
+
+export default class CustomObjectsExporter {
+  // Set type annotations
+  apiConfig: ApiConfigOptions
+  client: Client
+  logger: LoggerOptions
+  predicate: ?string
+
+  constructor(options: ExporterOptions, logger: LoggerOptions) {
+    if (!options.apiConfig)
+      throw new Error('The constructor must be passed an `apiConfig` object')
+    this.apiConfig = options.apiConfig
+    this.client = createClient({
+      middlewares: [
+        createAuthMiddlewareWithExistingToken(
+          options.accessToken ? `Bearer ${options.accessToken}` : ''
+        ),
+        createAuthMiddlewareForClientCredentialsFlow(this.apiConfig),
+        createUserAgentMiddleware({
+          libraryName: pkg.name,
+          libraryVersion: pkg.version,
+        }),
+        createHttpMiddleware({ host: this.apiConfig.apiUrl }),
+      ],
+    })
+
+    this.predicate = options.predicate
+
+    this.logger = {
+      error: () => {},
```

Comment:
You could extract that - maybe in another PR even - into a `utils/silent-logger` and then just

```js
this.logger = {
   ...silentLogger
   ...logger
}
```